src/pulp/solvers.py

```diff
@@ -2887,3 +2887,242 @@ def __init__(self, path=None, *args, **kwargs):
                 raise PulpSolverError('Use CHOCO_CMD if you want to set a path')
             # check that the file is executable
             CHOCO_CMD.__init__(self, path=self.pulp_choco_path, *args, **kwargs)
+
+########################################################################################################################
+##################################################     MOSEK-PuLP     ##################################################
+########################################################################################################################
+
+class MOSEK(LpSolver):
+    "PuLP interface to the Mosek solver"
+    try:
+        global mosek
+        import mosek 
+    except ImportError:
+        def available(self):
+            return False 
+        def actualSolve(self, lp, callback = None):
+            raise PulpSolverError("MOSEK : Not Available")
+    else:
+        def __init__(self, mip = True, msg = True, mosek_pars = {}, task_file_name = "", sol_type = mosek.soltype.bas):
+            self.mip = mip
+            self.msg = msg
+            self.task_file_name = task_file_name
+            self.solution_type = sol_type
+            self.mosek_pars_dict = mosek_pars
+
+        def available(self):
+            print('MOSEK : Available')
+            return(True)
+
+        def streampunk(self, text):
+            sys.stdout.write(text)
+            sys.stdout.flush()
+
+        def buildSolverModel(self, lp, inf=1e20, LpObjSenses = {LpMaximize:-1, LpMinimize:1}):
+            self.cons = lp.constraints
+            self.numcons = len(self.cons)
+            self.cons_dict = {}
+            i = 0
+            for c in self.cons:
+                self.cons_dict[c] = i
+                i = i+1
+            self.vars = list(lp.variables())
+            self.numvars = len(self.vars)
+            self.var_dict = {}
+            #Checking for repeated names
+            repeated_names = {}
+            for v in self.vars:
```

Comment:
in the master branch, the `lp` object has a `checkDuplicateVars` method that does just this.